### DISCUSSION TITLE
Decouple versioning of the core Meteor Typescript package from Typescript itself because they do not use semantic versioning

### DISCUSSION THREAD

**Discussion by perbergland** [meteor/meteor#12490] [Feature Requests]
The core Typescript package in Meteor follows the version of Typescript, but that is a mistake since Typescript-npm does not use semantic versioning.

This caused a lot of unnecessary problems with packages suddenly becoming incompatible with the 4.0 release (and meteor has no way to force version resolution).

Please ensure that for the upcoming Typescript 5.0 release that the Typescript package version is decoupled from the Typescript npm package version. I suggest that you simply move to version 4.10.x from 4.9.x and then 4.11 etc.

**Comment by radekmie** [comment]
> [...] (and meteor has no way to force version resolution).

If I recall correctly, putting a `!` at the end of version forced it. And if you ask me, I'd rather keep the versioning as it is, as it eliminates one more version to check while upgrading.

**Comment by perbergland** [comment]
I was around with all the pain caused by the 3.9 => 4.0 update where packages were suddenly unnecessarily deemed incompatible by meteor and would like to avoid that pain this time.
